src/test/amazonqFeatureDev/controllers/chat/controller.test.ts

```diff
@@ -158,23 +161,84 @@ describe('Controller', () => {
+            const expectedSourceRoot = path.join(controllerSetup.workspaceFolder.uri.fsPath, 'src')
+            await modifyDefaultSourceFolder(expectedSourceRoot)
+            assert.strictEqual(session.config.sourceRoot, expectedSourceRoot)
+            assert.strictEqual(session.config.workspaceRoot, controllerSetup.workspaceFolder.uri.fsPath)
+        })
+    })
+
+    describe('processChatItemVotedMessage', () => {
+        async function processChatItemVotedMessage(vote: 'upvote' | 'downvote') {
+            const initialState = new PrepareRefinementState(
+                {
+                    conversationId: conversationID,
+                    proxyClient: new FeatureDevClient(),
+                    sourceRoot: '',
+                    workspaceRoot: '',
+                },
+                '',
+                tabID
+            )
+            const newSession = await createSession({
+                messenger: controllerSetup.messenger,
+                sessionState: initialState,
                 conversationID,
                 tabID,
                 uploadID,
             })
-            sinon.stub(vscode.workspace, 'getWorkspaceFolder').returns(controllerSetup.workspaceFolder)
-            const expectedSourceRoot = path.join(controllerSetup.workspaceFolder.uri.fsPath, 'src')
-            const modifiedSourceFolderSession = await modifyDefaultSourceFolder(
-                controllerSetup.emitters,
-                controllerSetup.session,
-                expectedSourceRoot
-            )
-            assert.strictEqual(modifiedSourceFolderSession.config.sourceRoot, expectedSourceRoot)
-            assert.strictEqual(
-                modifiedSourceFolderSession.config.workspaceRoot,
-                controllerSetup.workspaceFolder.uri.fsPath
-            )
+            const getSessionStub = sinon.stub(controllerSetup.sessionStorage, 'getSession').resolves(newSession)
+            controllerSetup.emitters.processChatItemVotedMessage.fire({
+                tabID,
+                messageID: '',
+                vote,
+            })
+
+            // Wait until the controller has time to process the event
+            await waitUntil(() => {
+                return Promise.resolve(getSessionStub.callCount > 0)
+            }, {})
+        }
+
+        it('incoming upvoted message sends telemetry', async () => {
+            await processChatItemVotedMessage('upvote')
+
+            const voteThumbsUpTelemetry = assertTelemetryCurried('amazonq_approachThumbsUp')
+            voteThumbsUpTelemetry({ amazonqConversationId: conversationID, result: 'Succeeded' })
+        })
+
+        it('incoming downvoted message sends telemetry', async () => {
+            await processChatItemVotedMessage('downvote')
+
+            const voteThumbsDownTelemetry = assertTelemetryCurried('amazonq_approachThumbsDown')
+            voteThumbsDownTelemetry({ amazonqConversationId: conversationID, result: 'Succeeded' })
```

Comment:
could use `assertTelemetry` directly instead of the "curried" form, because this code just immediately invoked the curried function.

same for the other cases.

I'd prefer if this change is made so that this doesn't get copy-pasted in many places.